sale_subscription_financial_risk/models/sale_subscription_financial_risk_cron.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (C) 2019 Amplex
+# Copyright (C) 2019 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from datetime import date, timedelta
+from dateutil.relativedelta import relativedelta
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class SaleSubscription(models.Model):
+    _inherit = 'sale.subscription'
+
+    # Calculates and returns a timedelta based on the customer's credit terms
+    def _get_delta_qty(self, qty, uom):
+        if uom == 'days':
```

Comment:
Instead of each individual comparison, you can do something like this  https://stackoverflow.com/questions/1649753/generating-a-dynamic-time-delta-python

But In the end it will be same, So you can Ignore this :smiley:

Reply:
Oh this is brilliant, I was trying to do something very similar to this initially but couldn't manage to get it working - thanks for the link :)

Reply:
```
def _get_delta_qty(self, qty, uom):
    uom_list = ["days", "weeks", "months", "years"]
    if uom in uom_list:
        kwargs = {uom: qty}
        return relativedelta(**kwargs)

``

Reply:
Earlier it was `timedelta` for days and `relativedelta` for other uom
https://github.com/ursais/osi-addons/pull/186/commits/e28ecd071eef8bfdf3a6de83e515dfce279722d0#diff-1d711372a7d02d917bcca7ed6574476aL16

Just to be sure you are not breaking something while Improving this

Reply:
I tested the change in the interpreter before applying, no worries :)